### ISSUE TITLE
Failed to load TextAsset "cute_robot" as a Rive file.

### ISSUE THREAD

**Issue by Adeptus-Astartes** [rive-app/rive-unity#25]
SDK Version: 0.1.96. Platform: Android

```
MemoryManager: Using 'Dynamic Heap' Allocator.
03-20 16:23:33.200 26180 26226 I Unity   : SystemInfo CPU = ARM64 FP ASIMD AES, Cores = 8, Memory = 7200mb
03-20 16:23:33.200 26180 26226 I Unity   : SystemInfo ARM big.LITTLE configuration: 4 big (mask: 0xf0), 4 little (mask: 0xf)
03-20 16:23:33.200 26180 26226 I Unity   : ApplicationInfo com.DefaultCompany.RiveExample version 0.1
03-20 16:23:33.200 26180 26226 I Unity   : Built from '2021.3/staging' branch, Version '2021.3.9f1 (ad3870b89536)', Build type 'Release', Scripting Backend 'mono', CPU 'armeabi-v7a', Stripping 'Disabled'
03-20 16:23:33.312 26180 26226 I Unity   : Company Name: DefaultCompany
03-20 16:23:33.312 26180 26226 I Unity   : Product Name: Rive Example
03-20 16:23:35.908 26180 26226 I Unity   : Failed to load TextAsset "cute_robot" as a Rive file.
03-20 16:23:35.952 26180 26226 E Unity   : NullReferenceException: Object reference not set to an instance of an object
03-20 16:23:35.952 26180 26226 E Unity   :   at RiveTexture.Start () [0x00038] in <42b37d73b1e44baa92c4a58c82200896>:0 
03-20 16:23:35.952 26180 26226 E Unity   : 

```

**Comment by jleblancsw** [comment]
I'm also running into this problem. I'm thinking maybe the Rive plugin dll stuff isn't packaging or loading correctly into Android?
When it sees the Rive references in code maybe it's not seeing it?

**Comment by Adeptus-Astartes** [comment]
Hi, it was quite long time ago, maybe it's already fixed in latest
versions, Haven't check yet

пт, 21 черв. 2024 р. о 01:44 jleblancsw ***@***.***> пише:

**Comment by jleblancsw** [comment]
I figured it out. In my case Unity was building the app for Vulkan. I had to turn of "Auto Graphics API" in Unity's Project Settings->Player->Other Settings and remove Vulkan and use ES3

Also be warned the Rive plugin is hardcoded to get the Camera.main to display it correctly. You may need to edit that code or set your main camera to use the MainCamera tag if you only have one camera

**Comment by damzobridge** [comment]
This should be fixed from [v0.1.188](https://github.com/rive-app/rive-unity/releases/tag/v0.1.188)! Let us know if you still run into any issues with Android not loading Rive files.
